experimental/stats/metricregistry.go

```diff
@@ -250,9 +249,9 @@ func RegisterInt64Gauge(descriptor MetricDescriptor) *Int64GaugeHandle {
 }
 
 // snapshotMetricsRegistryForTesting snapshots the global data of the metrics
-// registry. Registers a cleanup function on the provided testing.T that sets
-// the metrics registry to its original state. Only called in testing functions.
-func snapshotMetricsRegistryForTesting(t *testing.T) {
+// registry. Returns a cleanup function that sets the metrics registry to its
+// original state.
+func snapshotMetricsRegistryForTesting() func() {
```

Comment:
If this isn't exported anyway, then can't we simply move it into an `_test.go` file?

Reply:
We can't do that since `internal.SnapshotMetricRegistryForTesting` is set to this unexported function (in an `init` function) and the variable in `internal` package is used by other tests and therefore needs to be set.